lib/SILOptimizer/Transforms/ArrayElementValuePropagation.cpp

```diff
@@ -249,6 +278,39 @@ bool ArrayAllocation::collectForwardableValues() {
 namespace {
 
 class ArrayElementPropagation : public SILFunctionTransform {
+
+  FuncDecl* lookupAppendFnDecl() {
+    auto &M = getFunction()->getModule();
+    auto &C = M.getASTContext();
+    auto appendFunctions = C.getArrayDecl()->lookupDirect(C.getIdentifier("append"));
+    FuncDecl *appendFnDecl;
+
+    for (auto candidateFn : appendFunctions) {
+      auto *fnDecl = dyn_cast_or_null<FuncDecl>(candidateFn);
+      if (fnDecl == NULL) {
+        continue;
+      }
+
+      auto paramLists = fnDecl->getParameterLists();
+      if (paramLists.size() != 2) {
+        continue;
+      }
+
+      auto secondParamList = paramLists[1];
+      if (secondParamList->size() != 1) {
+        continue;
+      }
+
+      auto paramName = secondParamList->get(0)->getName();
+      if (paramName == C.getIdentifier("newElement")) {
+        appendFnDecl = fnDecl;
+        break;
+      }
+    }
+
+    return appendFnDecl;
```

Comment:
Good call. I was thinking about doing that, but couldn't figure out how to do that with a `FuncDecl` at the time. I finally figured it out, though. Done.